src/components/backend/splitter.js

```diff
@@ -0,0 +1,104 @@
+import {CharacterTextSplitter} from "langchain/text_splitter";
+
+export class CharacterTextSplitterOffset
+  extends CharacterTextSplitter
+{
+    static lc_name() {
+        return "CharacterTextSplitterOffset";
+    }
+
+separator = "\n\n";
```

Comment:
The styling of this file doesn't look right e.g. indentation from this line is off, and length of indentation is mixed (2 vs 4). Maybe try the prettier tool added by @davidspu ?